INSTALLATION AND DATA IMPORTATION

In [3]:
#Installing and importing libraries for data importation
!pip install pandas



[notice] A new release of pip is available: 24.1.2 -> 24.2
[notice] To update, run: pip install --upgrade pip


In [2]:
#import libraries
import pandas as pd

In [6]:
#import dataset excel
pd.read_excel("diamonds_new.xlsx")


price  carat        cut color clarity  depth  table     x     y     z
0        326   0.23      Ideal     E     SI2   61.5   55.0  3.95  3.98  2.43
1        326   0.21    Premium     E     SI1   59.8   61.0  3.89  3.84  2.31
2        327   0.23       Good     E     VS1   56.9   65.0  4.05  4.07  2.31
3        334   0.29    Premium     I     VS2   62.4   58.0  4.20  4.23  2.63
4        335   0.31       Good     J     SI2   63.3   58.0  4.34  4.35  2.75
...      ...    ...        ...   ...     ...    ...    ...   ...   ...   ...
53915   2757   0.72      Ideal     D     SI1   60.8   57.0  5.75  5.76  3.50
53916   2757   0.72       Good     D     SI1   63.1   55.0  5.69  5.75  3.61
53917   2757   0.70  Very Good     D     SI1   62.8   60.0  5.66  5.68  3.56
53918   2757   0.86    Premium     H     SI2   61.0   58.0  6.15  6.12  3.74
53919   2757   0.75      Ideal     D     SI2   62.2   55.0  5.83  5.87  3.64

[53920 rows x 10 columns]

In [3]:
#import the dataset as a pandas dataframe
class1=pd.read_excel("diamonds_new.xlsx")

#View the new dataframe called class1
class1

price  carat        cut color clarity  depth  table     x     y     z
0        326   0.23      Ideal     E     SI2   61.5   55.0  3.95  3.98  2.43
1        326   0.21    Premium     E     SI1   59.8   61.0  3.89  3.84  2.31
2        327   0.23       Good     E     VS1   56.9   65.0  4.05  4.07  2.31
3        334   0.29    Premium     I     VS2   62.4   58.0  4.20  4.23  2.63
4        335   0.31       Good     J     SI2   63.3   58.0  4.34  4.35  2.75
...      ...    ...        ...   ...     ...    ...    ...   ...   ...   ...
53915   2757   0.72      Ideal     D     SI1   60.8   57.0  5.75  5.76  3.50
53916   2757   0.72       Good     D     SI1   63.1   55.0  5.69  5.75  3.61
53917   2757   0.70  Very Good     D     SI1   62.8   60.0  5.66  5.68  3.56
53918   2757   0.86    Premium     H     SI2   61.0   58.0  6.15  6.12  3.74
53919   2757   0.75      Ideal     D     SI2   62.2   55.0  5.83  5.87  3.64

[53920 rows x 10 columns]

## PROBABILITY OF DIAMONDS DATASET ##

In [12]:
class1["cut"].unique()

array(['Ideal', 'Premium', 'Good', 'Very Good', 'Fair'], dtype=object)

In [13]:
#First we get the different number cuts of diamonds in the dataset
cut_total = class1['cut'].count()
#View the result
cut_total

np.int64(53920)

In [14]:
#Then get the count of just "premium" diamonds
premium = class1[class1['cut'] == "Premium"]['cut'].count()
#View result
premium

np.int64(13789)

In [15]:
#Now calculate the probability of having a Premium diamond
probability=premium/cut_total
#View result
probability

np.float64(0.2557307121661721)

The probability is 25.57%

**Conditional Probability**

What is the probability of having a Premium diamond that is "J" coloured?

In [16]:
#First filter the  data for only premium diamonds that are J coloured
filtered_data = (class1['cut'] == 'Premium') & (class1['color'] == 'J')

In [17]:
#number of outcomes of selecting Premium and J
premium_j_count = class1[filtered_data]["cut"].count()
#View result
premium_j_count

np.int64(808)

In [18]:
#Total outcomes of Premium 

### Probability of getting a fair diamondD

In [19]:
fair_diamonds=class1[class1['cut']=="Fair"]['cut'].count()
fair_diamonds

np.int64(1609)

In [20]:
fairprobability=fair_diamonds/cut_total
fairprobability

np.float64(0.029840504451038575)